appveyor-windows-build-go.yml

```diff
@@ -0,0 +1,79 @@
+version: 1.0.{build}
+image: Windows Server 2019
+build: off
+
+# Change the clone folder to somewhere in "D:\" because this is shared by default with Docker. We need this to mount folders.
+clone_folder: D:\source
+
+environment:
+  AWS_DEFAULT_REGION: us-east-1
+  SAM_CLI_DEV: 1
+  APPVEYOR_CI_OVERRIDE: 1
+
+  # In Windows, tempdir is usually in C:\. But in AppVeyor only D:\ is shared in Docker. 
+  # Therefore change TEMPDIR it to D: for invoke tests to work. Python uses $TMPDIR envvar
+  # to find root of tempdir
+  TMPDIR: D:\tmp
+  TEMP: D:\tmp
+  TMP: D:\tmp
+
+  # MSI Installers only use Py3.6.6. It is sufficient to test with this version here.
+  PYTHON_HOME: "C:\\Python36-x64"
+  PYTHON_SCRIPTS: "C:\\Python36-x64\\Scripts"
+  PYTHON_EXE: "C:\\Python36-x64\\python.exe"
+  PYTHON_VERSION: '3.6.8'
+  PYTHON_ARCH: '64'
+  HOME: 'C:\Users\appveyor'
+  HOMEDRIVE: 'C:'
+  HOMEPATH: 'C:\Users\appveyor'
+
+init:
+  # Uncomment this for RDP
+  - ps: iex ((new-object net.webclient).DownloadString('https://raw.githubusercontent.com/appveyor/ci/master/scripts/enable-rdp.ps1'))
+  - ps: gcim Win32_Processor | % { "$($_.NumberOfLogicalProcessors) logical CPUs" }
+  - ps: gcim Win32_OperatingSystem | % { "$([int]($_.TotalVisibleMemorySize/1mb)) Gb" }
+
+
+install:
+
+    # Make sure the temp directory exists for Python to use.
+    - ps: "mkdir -Force D:\\tmp"
+    - "SET PATH=%PYTHON_HOME%;%PATH%"
+    - "echo %PYTHON_HOME%"
+    - "echo %PATH%"
+    - "python --version"
+
+    # Upgrade setuptools, wheel and virtualenv
+    - "python -m pip install --upgrade setuptools wheel virtualenv"
+
+    # Create new virtual environment with chosen python version and activate it
+    - "python -m virtualenv venv"
+    - "venv\\Scripts\\activate"
+    - "python --version"
+
+    # Actually install SAM CLI's dependencies
+    - "pip install -e \".[dev]\""
+
+    # Switch to Docker Linux containers
+    - ps: Switch-DockerLinux
+
+    # Check Go Install
+    - "SET PATH=C:\\Go\\bin;%PATH%"
+    - "echo %PATH%"
+    - "dir C:\\Go\\bin"
+    - "go version"
+    - "go env"
+
+    # Echo final Path
+    - "echo %PATH%"
+
+test_script:
+    # Reactivate virtualenv before running tests
+    - "venv\\Scripts\\activate"
+    - "docker system prune -a -f"
+    - "pytest -vv tests/integration/buildcmd/test_build_cmd.py -k test_with_go"
+    - "pytest -vv tests/integration/buildcmd/test_build_cmd.py -k test_go_must_fail_with_container"
```

Comment:
Does this tests need docker installed? I don't remember the code paths but if we don't need docker installed to test this, then we can run it in the 'java-ruby-inprocess-build' appveyor project.